gnome-image-installer/util/gis-unattended-config.c

```diff
@@ -118,6 +124,53 @@ gis_unattended_config_class_init (GisUnattendedConfigClass *klass)
   g_object_class_install_properties (object_class, N_PROPERTIES, props);
 }
 
+static gboolean
+key_file_get_optional_string (GKeyFile    *key_file,
+                              const gchar *group_name,
+                              const gchar *key,
+                              gchar      **value_out,
+                              GError     **error)
+{
+  g_autoptr(GError) local_error = NULL;
+  gchar *value;
+
+  value = g_key_file_get_string (key_file, group_name, key, &local_error);
+  if (value == NULL)
+    {
+      if (g_error_matches (local_error, G_KEY_FILE_ERROR,
+                           G_KEY_FILE_ERROR_GROUP_NOT_FOUND) ||
+          g_error_matches (local_error, G_KEY_FILE_ERROR,
+                           G_KEY_FILE_ERROR_KEY_NOT_FOUND))
+        {
+          return TRUE;
```

Comment:
I suspect you should be setting `*value_out = NULL` in this case.

Reply:
I reworked this function to make this case be covered by the regular success path.